opencensus/metrics/export/value.py

```diff
@@ -86,5 +89,148 @@ class ValueSummary(Value):
     :type value: summary
     :param value: the value in summary.
     """
+
     def __init__(self, value):
         super(ValueSummary, self).__init__(value)
+
+
+class Exemplar(object):
+    """An example point to annotate a given value in a bucket.
+
+    Exemplars are example points that may be used to annotate aggregated
+    Distribution values. They are metadata that gives information about a
+    particular value added to a Distribution bucket.
+
+    :type value: double
+    :param value: Value of the exemplar point, determines which bucket the
+    exemplar belongs to.
+
+    :type timestamp: str
+    :param timestamp: The observation (sampling) time of the exemplar value.
+
+    :type attachments: dict(str, str)
+    :param attachments: Contextual information about the example value.
+    """
+
+    def __init__(self, value, timestamp, attachments):
+        self._value = value
+        self._timestamp = timestamp
+        self._attachments = attachments
+
+    @property
+    def value(self):
+        return self._value
+
+    @property
+    def timestamp(self):
+        return self._timestamp
+
+    @property
+    def attachments(self):
+        return self._attachments
+
+
+class Bucket(object):
+    """A bucket of a histogram.
+
+    :type count: int
+    :param count: The number of values in each bucket of the histogram.
+
+    :type exemplar: Exemplar
+    :param exemplar: Optional exemplar for this bucket, omit if the
+    distribution does not have a histogram.
+    """
+
+    def __init__(self, count, exemplar):
+        self._count = count
+        self._exemplar = exemplar
+
+    @property
+    def count(self):
+        return self._count
+
+    @property
+    def exemplar(self):
+        return self._exemplar
+
+
+def window(ible, width):
+    win = []
+    for x in ible:
+        win = (win + [x])[-width:]
+        if len(win) == width:
+            yield win
+
+
+class ValueDistribution(Value):
+    """Summary statistics for a population of values.
+
+    Distribution contains summary statistics for a population of values. It
+    optionally contains a histogram representing the distribution of those
+    values across a set of buckets.
+
+    :type count: int
+    :param count: The number of values in the population.
+
+    :type sum_: float
+    :param sum_: The sum of the values in the population.
+
+    :type sum_of_squared_deviation: float
+    :param sum_of_squared_deviation: The sum of squared deviations from the
+    mean of the values in the population.
+
+    :type bucket_bounds: list(float)
```

Comment:
Please add TODO here to change this to ```BucketOptions``` as per new [metric.proto](https://github.com/census-instrumentation/opencensus-proto/blob/master/src/opencensus/proto/metrics/v1/metrics.proto#L204).